ttnn/cpp/ttnn/run_operation_inl.hpp

```diff
@@ -232,10 +232,7 @@ void launch_op(
                         insert_buffer_and_shape_for_device(target_device, *local_tensor, *output_tensor);
                         int num_workers_completed = (output_tensor->tensor_attributes->num_workers_completed)++;
                         if (not num_workers_completed) {
-                            output_tensor->tensor_attributes->shape = local_tensor->tensor_attributes->shape;
-                            output_tensor->tensor_attributes->dtype = local_tensor->tensor_attributes->dtype;
-                            output_tensor->tensor_attributes->layout = local_tensor->tensor_attributes->layout;
-                            output_tensor->tensor_attributes->tile = local_tensor->tensor_attributes->tile;
+                            output_tensor->set_tensor_spec(local_tensor->tensor_spec());
```

Comment:
not a good public method, would like to discuss this.

Reply:
I agree. It's used because the shape of the tensor isn't initially know. I believe we can fix this after all ops are migrated to `compute_output_specs`.
I do believe it beats setting 5 fields by hand though.

Reply:
If you're planning to get rid of the method, there are ways of trying to keep the usage to a a minimum (private, protected, friend, deprecated).